src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1000,603 +946,315 @@ private ImmutableArray<Symbol> CreateMembers()
+                TupleUnderlyingType = underlyingType;
+                Locations = ImmutableArray<Location>.Empty;
             }
-        }
 
-        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences
-        {
-            get
+            internal TupleUncommonData(NamedTypeSymbol underlyingType, ImmutableArray<string> elementNames,
+                ImmutableArray<Location> elementLocations, ImmutableArray<bool> errorPositions, ImmutableArray<Location> locations)
+                : this(underlyingType)
             {
-                return GetDeclaringSyntaxReferenceHelper<CSharpSyntaxNode>(_locations);
+                ElementNames = elementNames;
+                ElementLocations = elementLocations;
+                ErrorPositions = errorPositions;
+                Locations = locations.IsDefault ? ImmutableArray<Location>.Empty : locations;
             }
-        }
 
-        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverrideOpt = null)
-        {
-            if ((comparison & TypeCompareKind.IgnoreTupleNames) != 0)
+            internal bool EqualsIgnoringTupleUnderlyingType(TupleUncommonData? other)
             {
-                if (t2?.IsTupleType == true)
+                if (other is null && this.ElementNames.IsDefault && this.ElementLocations.IsDefault && this.ErrorPositions.IsDefault)
                 {
-                    t2 = t2.TupleUnderlyingType;
+                    return true;
                 }
 
-                return _underlyingType.Equals(t2, comparison, isValueTypeOverrideOpt);
-            }
+                return other is object &&
+                    this.ElementNames == other.ElementNames
+                    && compareElementLocations(other)
+                    && this.ErrorPositions == other.ErrorPositions;
 
-            return this.Equals(t2 as TupleTypeSymbol, comparison, isValueTypeOverrideOpt);
-        }
+                bool compareElementLocations(TupleUncommonData other)
+                {
+                    if (this.ElementLocations.IsDefault && other.ElementLocations.IsDefault)
+                    {
+                        return true;
+                    }
 
-        internal bool Equals(TupleTypeSymbol other)
-        {
-            return Equals(other, TypeCompareKind.ConsiderEverything);
-        }
+                    if (this.ElementLocations.IsDefault != other.ElementLocations.IsDefault)
+                    {
+                        return false;
+                    }
 
-        private bool Equals(TupleTypeSymbol other, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverrideOpt)
-        {
-            if (ReferenceEquals(this, other))
-            {
-                return true;
+                    return this.ElementLocations.SequenceEqual(other.ElementLocations, (l1, l2) => (l1 is object && l1.Equals(l2)) || l2 is null);
```

Comment:
>, (l1, l2) => (l1 is object && l1.Equals(l2)) || l2 is null [](start = 85, length = 59)

`, (l1, l2) => object.Equals(l1, l2)` or just use the default comparer which should do the same thing.